File: backend/semgrep_rules/tests/javascript/kisa_sw_44_cross_session_data_exposure.js
let currentProfile;
function unsafe(req) {
  // ruleid: kisa.sw44.javascript.request-data-to-module-global-review
  currentProfile = req.query.profile;
  return currentProfile;
}
function safe(req) {
  // ok: kisa.sw44.javascript.request-data-to-module-global-review
  return { profile: req.query.profile };
}

File: backend/semgrep_rules/tests/python/kisa_sw_44_cross_session_data_exposure.py
current_user_data = None
def unsafe(request):
    global current_user_data
    # ruleid: kisa.sw44.python.request-data-to-global-state-review
    current_user_data = request.GET.get("profile")
    return current_user_data
def safe(request):
    # ok: kisa.sw44.python.request-data-to-global-state-review
    return {"profile": request.GET.get("profile")}
